### PR TITLE
Simplifying the Fields (Controls) interface

### PR DESCRIPTION
**Pull request by mistercrunch**
* `FieldContainer` now wraps `ControlLabelWithTooltip`, as opposed to each `Field` doing so
* Moved to support a single `SelectField` as opposed to multiple flavors of it that had specific prop assignment logic in the `FieldSet` component. Moved those to become explicit props in the `fields` configuration object.

@ascott @vera-liu 

this is a part of https://github.com/airbnb/superset/pull/1788

### CODE REVIEW

**Comment by vera-liu** [comment]
what's the benefit of using one select field with props in fields instead of having one general SelectField that fits different types?

**Comment by vera-liu** [comment]
it seems that FieldSet wasn't renamed to FieldContainer, should we delete it from the description?

File: superset/assets/javascripts/explorev2/stores/fields.js [hunk 1]
```diff
@@ -55,15 +45,19 @@ export const fields = {
   },
 
   metrics: {
-    type: 'SelectMultipleSortableField',
+    type: 'SelectField',
+    multi: true,
+    freeForm: false,
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/stores/fields.js]: i like how you've simplified the select field type to `SelectField` and added `multi` and `freeform` keys here, but i don't see where the changes are in the `SelectField` component to accommodate these new keys. are they in a different PR?

Comment (ascott) [inline superset/assets/javascripts/explorev2/stores/fields.js]: ah, nevermind, i see the changes now!

File: superset/assets/javascripts/explorev2/components/FieldSet.jsx [hunk 2]
```diff
@@ -18,68 +27,22 @@ const propTypes = {
...
-  }
-
-  renderTextField() {
-    return (
-      <TextField
-        {...this.props}
-      />);
-  }
-
+export default class FieldSet extends React.PureComponent {
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/FieldSet.jsx]: do you mean for this file/component to be called `FieldContainer` now?

File: superset/assets/javascripts/explorev2/components/FieldSet.jsx [hunk 3]
```diff
@@ -18,68 +27,22 @@ const propTypes = {
...
-    } else if (Object.keys(selectProps).includes(type)) {
-      field = this.renderSelectField(selectProps[type]);
-    } else if (['TextField', 'IntegerField'].includes(type)) {
-      field = this.renderTextField();
-    } else if (type === 'TextAreaField') {
-      field = this.renderTextAreaField();
-    }
-
-    return field;
+    const FieldClass = fieldMap[this.props.type];
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/FieldSet.jsx]: nice simplification here. what do you think of calling this `FieldType` rather than `FieldClass`?

File: superset/assets/javascripts/explorev2/components/FieldSet.jsx (hunk 2 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/FieldSet.jsx]: oh I think it's in the next PR, I'll remove the bullet from the PR description

**Comment by mistercrunch** [comment]
@vera-liu, I thought using one `SelectField` instead of a collection of "virtual fields" that infer props simplifies the logic and makes things more simple/traceable. With this new code you can look at the `metrics` config array and know the correspondence of which component will be called, exactly which props will be passed, and which props will be evaluated from state.

Without that simplicity, someone might look for the virtual fields in the components folder and not find them, have to grep to find the logic and understand the props association. I also want to prevent more logic from slipping into that layer as things are already pretty hard to follow already when simplified.

If that config gets outs of control in the future, we may want to revisit the idea of "virtual field" associated with predefined sets of props, but for now I'd rather have something simple and explicit. If we were to add this logic is should be generic and live close to the `metrics` array, not in `FieldSet.js`.

**Comment by ascott** [comment]
these changes lgtm.
